bindings/web/diffie_hellman.cc

```diff
@@ -0,0 +1,177 @@
+#include <emscripten.h>
+#include <emscripten/bind.h>
+#include <emscripten/val.h>
+
+#include <sstream>
+#include <iostream>
+#include <iomanip>
+
+#include <openssl/evp.h>
+#include <openssl/ec.h>
+#include <openssl/pem.h>
+#include <openssl/rand.h>
+#include <openssl/err.h>
+#include <openssl/sha.h>
+
+using namespace emscripten;
+
+#define ERROR_NULL(VAR, FUN)                                       \
+    if(_return.empty())                                            \
+    {                                                              \
+        VAR = FUN;                                                 \
+        if(VAR == NULL)                                            \
+            _return = "error-openssl:" + std::to_string(__LINE__); \
+    }
+
+#define ERRORC(FUN)                                                    \
+    {                                                                  \
+        if(_return.empty())                                            \
+        {                                                              \
+            size_t a = (size_t)FUN;                                    \
+            if(a != 1)                                                 \
+                _return = "error-openssl:" + std::to_string(__LINE__); \
+        }                                                              \
+    }
```

Comment:
The _return variable should not be hardcoded into the macro. Code using this is not so much readable
Also improve names

Reply:
How I should improve this? I need macros in order to correctly `free` evp variables, because for each time an error occurs it should `free` variables allocated until the error. In this way `free` are called only one time.

Reply:
I see but this macro as written can't be used if you don't declare a `_return` variable and this is not good for readability because the variable seems to be unused. It should be an argument, or (even better) since this is C++ you could employ a `static constexpr` function.

Also what does ERRORC stands for?

Reply:
ERRORC stands for ERROR_CHECK but it was a temporary name I forgot to rename. It is used for checking EVP functions which returns successfully when they are equal to 1. Could I call it `ERROR_EVP` ?

Reply:
Yes, it would be more clear